src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -85,40 +85,60 @@ static PSVersionInfo()
             //    when powershell is built from a release tag for RTM:
-            //      productVersion = '6.0.0 SHA: f1ec9...'                    convert to GitCommitId = 'v6.0.0'
-            //                                                                           PSVersion   = '6.0.0'
-            string rawGitCommitId;
-            string mainVersion = ProductVersion.Substring(0, ProductVersion.IndexOf(' '));
+            //      productVersion = '7.3.0 SHA: f1ec9...'                    convert to GitCommitId = 'v7.3.0'
+            //                                                                           PSVersion   = '7.3.0'
+            int index = ProductVersion.IndexOf(' ');
+            ReadOnlySpan<char> mainVersion = ProductVersion.AsSpan(0, index);
 
-            if (ProductVersion.Contains(" Commits: "))
-            {
-                rawGitCommitId = ProductVersion.Replace(" Commits: ", "-").Replace(" SHA: ", "-g");
-            }
-            else
-            {
-                rawGitCommitId = mainVersion;
-            }
+            string rawGitCommitId = ProductVersion.AsSpan(index).StartsWith(" Commits: ")
+                ? "v" + ProductVersion.Replace(" Commits: ", "-").Replace(" SHA: ", "-g")
+                : string.Concat("v".AsSpan(), mainVersion);
 
-            s_psSemVersion = new SemanticVersion(mainVersion);
-            s_psVersion = (Version)s_psSemVersion;
+            var r = FastParsePSVersion(mainVersion);
+            s_psSemVersion = r.preReleaseLabel is null
+                ? new SemanticVersion(r.major, r.minor, r.patch)
+                : new SemanticVersion(r.major, r.minor, r.patch, r.preReleaseLabel, buildLabel: null);
 
-            s_psVersionTable[PSVersionInfo.PSVersionName] = s_psSemVersion;
-            s_psVersionTable[PSVersionInfo.PSEditionName] = PSEditionValue;
+            s_psVersion = (Version)s_psSemVersion;
+            s_psVersionTable[PSVersionName] = s_psSemVersion;
+            s_psVersionTable[PSEditionName] = PSEditionValue;
             s_psVersionTable[PSGitCommitIdName] = rawGitCommitId;
             s_psVersionTable[PSCompatibleVersionsName] = new Version[] { s_psV1Version, s_psV2Version, s_psV3Version, s_psV4Version, s_psV5Version, s_psV51Version, s_psV6Version, s_psV61Version, s_psV62Version, s_psV7Version, s_psV71Version, s_psV72Version, s_psVersion };
-            s_psVersionTable[PSVersionInfo.SerializationVersionName] = new Version(InternalSerializer.DefaultVersion);
-            s_psVersionTable[PSVersionInfo.PSRemotingProtocolVersionName] = RemotingConstants.ProtocolVersion;
-            s_psVersionTable[PSVersionInfo.WSManStackVersionName] = GetWSManStackVersion();
+            s_psVersionTable[SerializationVersionName] = new Version(InternalSerializer.DefaultVersion);
+            s_psVersionTable[PSRemotingProtocolVersionName] = RemotingConstants.ProtocolVersion;
+            s_psVersionTable[WSManStackVersionName] = GetWSManStackVersion();
```

Comment:
You know, `hashtable` isn't sealed (and the relevant bits are virtual). I wonder if it makes sense to subclass it and make some of these lazily init.

(not really review feedback, just food for thought)

Reply:
I think it's doable. Will submit a separate PR for that.

Reply:
> I think it's doable. Will submit a separate PR for that.

No need to this work if we will use SG and put all that are used at startup to separate helper to avoid initialization of PSVersion static class.

Reply:
You cannot eliminate the cost of initializing the Hashtable with SG. But we may not need to do it depending measurement after adopting SG. See my comment https://github.com/PowerShell/PowerShell/pull/18210#discussion_r988120904.

Reply:
> You cannot eliminate the cost of initializing the Hashtable with SG.

*could* also roll a `PSVariable` implementation that lazily inits on first `get_Value`

Reply:
yeah, that's what I planed in https://github.com/PowerShell/PowerShell/pull/18210#discussion_r988120904.
But after integrating the `FastParserPSVersion` with source generator, the static constructor of `PSVersionInfo` only takes about 2.5ms on my dev machine (see https://github.com/PowerShell/PowerShell/pull/15603#pullrequestreview-1132183159), so now it does seem to be slight overkill now.